src/components/User/UserProfile.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+
+// components
+import Text from '../Text/Text';
+import { WalletCopyIcon } from '../TransactionBlock/Icons';
+
+// hooks
+import { useEtherspot } from '../../hooks';
+
+// icons
+import { HiCheck } from 'react-icons/hi';
+
+// utils
+import { Theme } from '../../utils/theme';
+import { ENSNode } from 'etherspot';
+import Card from '../Card';
+
+// constants
+import { OPENLOGIN_STORE } from '../../constants/storageConstants';
+
+const UserProfile = () => {
+  const { accountAddress, getENSNode } = useEtherspot();
+
+  const theme: Theme = useTheme();
+
+  const [copiedAddress, setCopiedAddress] = useState(false);
+  const [ensNode, setEnsNode] = useState<string | undefined>(undefined);
+
+  const openLoginStoreString = localStorage.getItem(OPENLOGIN_STORE);
+  const openLoginStore = openLoginStoreString && JSON.parse(openLoginStoreString);
+  const email = openLoginStore?.email ?? '';
+
+  const onCopy = async (valueToCopy: string) => {
+    try {
+      await navigator.clipboard.writeText(valueToCopy);
+      setCopiedAddress(true);
+      setTimeout(() => setCopiedAddress(false), 10000);
```

Comment:
as for the operations, we can pass callbacks to the geric function that'll execute `onSuccess`, `onFaliure` something like this
```tsx
export const onCopy = async (valueToCopy: string, onSuccess?: () => void, onFaliure?: () => void) => {
  try {
      await navigator.clipboard.writeText(valueToCopy);
      if(onSuccess) onSuccess();
      }
  catch (e) {
      if(onFaliure) return onFaliure();
      alert('Unable to copy');
    }
}
```

Reply:
yes makes sense, will execute this.